Transmission 4.0.2 Qt client compile issue in Ubuntu 20.04

### What is the issue?

The previous 4.0.0 and 4.0.1 were building successfully, but failed in the new 4.0.2 release with same debian/rules.

It built successfully in Ubuntu 22.04+, but failed in 20.04. Is there any dependency version change? Please help!

Build flags:
`mkdir build && cd build && cmake -DCMAKE_INSTALL_PREFIX:PATH=/usr -DENABLE_QT=ON -DENABLE_CLI=ON ../`

Build dependencies:
```
autotools-dev,
               debhelper,
               dh-autoreconf,
               cmake,
               intltool,                     
               libappstream-glib-dev,
               libappstreamqt-dev,
               libappstream-dev,
               appstream,                   
               libayatana-appindicator3-dev,
               libcurl4-openssl-dev | libcurl4-dev | libcurl-dev,
               libevent-dev (>= 2.0),
               libglib2.0-dev,
               libgtk-3-dev,
               libgtkmm-3.0-dev,
               libminiupnpc-dev (>= 1.9),
               libnatpmp-dev (>= 20110808),
               libnotify-dev (>= 0.7),
               libssl-dev,
               libsystemd-dev [linux-any],
               qt5-qmake,
               qt5-qmake:native,
               qtbase5-dev,
               qttools5-dev-tools,
               qttools5-dev,
               libqt5svg5-dev,
               zlib1g-dev
```

And here's the ERROR:

```
[ 79%] Building CXX object qt/CMakeFiles/transmission-qt.dir/DBusInteropHelper.cc.o
[ 79%] Building CXX object qt/CMakeFiles/transmission-qt.dir/DetailsDialog.cc.o
[ 79%] Building CXX object gtk/CMakeFiles/transmission-gtk.dir/FilterBar.cc.o
[ 79%] Building CXX object qt/CMakeFiles/transmission-qt.dir/FaviconCache.cc.o
/<<PKGBUILDDIR>>/qt/DetailsDialog.cc: In member function ‘void DetailsDialog::onAddTrackerClicked()’:
/<<PKGBUILDDIR>>/qt/DetailsDialog.cc:1395:66: error: no matching function for call to ‘QList<QString>::QList(<brace-enclosed initializer list>)’
 1395 |                 QList<QString>{ std::begin(urls), std::end(urls) });
      |                                                                  ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtCore/qhash.h:46,
                 from /usr/include/x86_64-linux-gnu/qt5/QtCore/qshareddata.h:46,
                 from /usr/include/x86_64-linux-gnu/qt5/QtCore/qdatetime.h:46,
                 from /usr/include/x86_64-linux-gnu/qt5/QtCore/QDateTime:1,
                 from /<<PKGBUILDDIR>>/qt/DetailsDialog.cc:13:
```

Full buildlog: https://launchpadlibrarian.net/656184233/buildlog_ubuntu-focal-amd64.transmission_4.0.2-1build1~focal_BUILDING.txt.gz

### Which application of Transmission?

Qt app on Linux, BSD, etc.

### Which version of Transmission?

Transmission 4.0.2

@PandaJim thanks for the build log :bow: 

```
/<<PKGBUILDDIR>>/qt/DetailsDialog.cc:1395:66: error: no matching function for call to ‘QList<QString>::QList(<brace-enclosed initializer list>)’
 1395 |                 QList<QString>{ std::begin(urls), std::end(urls) });
      |                                                                  ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtCore/qhash.h:46,
                 from /usr/include/x86_64-linux-gnu/qt5/QtCore/qshareddata.h:46,
                 from /usr/include/x86_64-linux-gnu/qt5/QtCore/qdatetime.h:46,
                 from /usr/include/x86_64-linux-gnu/qt5/QtCore/QDateTime:1,
                 from /<<PKGBUILDDIR>>/qt/DetailsDialog.cc:13:
/usr/include/x86_64-linux-gnu/qt5/QtCore/qlist.h:163:12: note: candidate: ‘QList<T>::QList(std::initializer_list<_Tp>) [with T = QString]’
  163 |     inline QList(std::initializer_list<T> args)
      |            ^~~~~
/usr/include/x86_64-linux-gnu/qt5/QtCore/qlist.h:163:12: note:   candidate expects 1 argument, 2 provided
/usr/include/x86_64-linux-gnu/qt5/QtCore/qlist.h:156:12: note: candidate: ‘QList<T>::QList(QList<T>&&) [with T = QString]’
  156 |     inline QList(QList<T> &&other) Q_DECL_NOTHROW
      |            ^~~~~
/usr/include/x86_64-linux-gnu/qt5/QtCore/qlist.h:156:12: note:   candidate expects 1 argument, 2 provided
/usr/include/x86_64-linux-gnu/qt5/QtCore/qlist.h:809:22: note: candidate: ‘QList<T>::QList(const QList<T>&) [with T = QString]’
  809 | Q_OUTOFLINE_TEMPLATE QList<T>::QList(const QList<T> &l)
      |                      ^~~~~~~~
/usr/include/x86_64-linux-gnu/qt5/QtCore/qlist.h:809:22: note:   candidate expects 1 argument, 2 provided
/usr/include/x86_64-linux-gnu/qt5/QtCore/qlist.h:151:12: note: candidate: ‘QList<T>::QList() [with T = QString]’
  151 |     inline QList() Q_DECL_NOTHROW : d(const_cast<QListData::Data *>(&QListData::shared_null)) { }
      |            ^~~~~
/usr/include/x86_64-linux-gnu/qt5/QtCore/qlist.h:151:12: note:   candidate expects 0 arguments, 2 provided
make[3]: *** [qt/CMakeFiles/transmission-qt.dir/build.make:292: qt/CMakeFiles/transmission-qt.dir/DetailsDialog.cc.o] Error 1
make[3]: *** Waiting for unfinished jobs....
```

Yes, this line was changed in 8fc904617b. Looks like it's relying on QList::QList(InputIterator first, InputIterator last) which was added in Qt 5.14, and this build is being made with Qt 5.12.8.

--- a/.github/workflows/actions.yml
+++ b/.github/workflows/actions.yml
@@ -736,23 +736,22 @@ jobs:
       - name: Get Dependencies
         run: |
           set -ex
-          apk update
-          apk add \
+          sudo apt-get update
+          sudo apt-get install -y --no-install-recommends \
             ca-certificates \
+            clang \
             cmake \
-            curl-dev \
-            fmt-dev \
-            g++ \
-            gettext-dev \
-            git \
+            gettext \
+            libcurl4-openssl-dev \
+            libdeflate-dev \
             libevent-dev \
-            libpsl \
-            linux-headers \
-            miniupnpc-dev \
-            ninja \
-            npm \
-            pkgconfig \
-            xz
+            libfmt-dev \
+            libminiupnpc-dev \
+            libnatpmp-dev \
+            libpsl-dev \
+            libssl-dev \
+            ninja-build \
+            npm
       - name: Get Dependencies (GTK)
         if: ${{ needs.what-to-make.outputs.make-gtk == 'true' }}
         run: apt-get install -y --no-install-recommends libglibmm-2.4-dev libgtkmm-3.0-dev

--- a/.github/workflows/actions.yml
+++ b/.github/workflows/actions.yml
@@ -587,8 +587,8 @@ jobs:
       - name: Get Dependencies
         run: |
           set -ex
-          sudo apt-get update
-          sudo apt-get install -y --no-install-recommends \
+          apt-get update
+          apt-get install -y --no-install-recommends \
             ca-certificates \
             cmake \
             g++ \
@@ -607,10 +607,10 @@ jobs:
             xz-utils
       - name: Get Dependencies (GTK)
         if: ${{ needs.what-to-make.outputs.make-gtk == 'true' }}
-        run: sudo apt-get install -y --no-install-recommends libglibmm-2.4-dev libgtkmm-3.0-dev
+        run: apt-get install -y --no-install-recommends libglibmm-2.4-dev libgtkmm-3.0-dev
       - name: Get Dependencies (Qt)
         if: ${{ needs.what-to-make.outputs.make-qt == 'true' }}
-        run: sudo apt-get install -y --no-install-recommends qtbase5-dev libqt5svg5-dev qttools5-dev
+        run: apt-get install -y --no-install-recommends qtbase5-dev libqt5svg5-dev qttools5-dev
       - name: Get Source
         uses: actions/download-artifact@v3
         with:
@@ -754,10 +754,10 @@ jobs:
             npm
       - name: Get Dependencies (GTK)
         if: ${{ needs.what-to-make.outputs.make-gtk == 'true' }}
-        run: apt-get install -y --no-install-recommends libglibmm-2.4-dev libgtkmm-3.0-dev
+        run: sudo apt-get install -y --no-install-recommends libglibmm-2.4-dev libgtkmm-3.0-dev
       - name: Get Dependencies (Qt)
         if: ${{ needs.what-to-make.outputs.make-qt == 'true' }}
-        run: apt-get install -y --no-install-recommends qtbase5-dev libqt5svg5-dev qttools5-dev
+        run: sudo apt-get install -y --no-install-recommends qtbase5-dev libqt5svg5-dev qttools5-dev
       - name: Get Source
         uses: actions/download-artifact@v3
         with:

--- a/.github/workflows/actions.yml
+++ b/.github/workflows/actions.yml
@@ -722,3 +722,78 @@ jobs:
         with:
           name: binaries-${{ github.job }}
           path: pfx/**/*
+
+  ubuntu-20-04-from-tarball:
+    needs: [ make-source-tarball, what-to-make ]
+    if: ${{ needs.what-to-make.outputs.make-cli == 'true' || needs.what-to-make.outputs.make-daemon == 'true' || needs.what-to-make.outputs.make-gtk == 'true' || needs.what-to-make.outputs.make-qt == 'true' || needs.what-to-make.outputs.make-tests == 'true' || needs.what-to-make.outputs.make-utils == 'true' }}
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Show Configuration
+        run: |
+          echo '${{ toJSON(needs) }}'
+          echo '${{ toJSON(runner) }}'
+          cat /etc/os-release
+      - name: Get Dependencies
+        run: |
+          set -ex
+          apk update
+          apk add \
+            ca-certificates \
+            cmake \
+            curl-dev \
+            fmt-dev \
+            g++ \
+            gettext-dev \
+            git \
+            libevent-dev \
+            libpsl \
+            linux-headers \
+            miniupnpc-dev \
+            ninja \
+            npm \
+            pkgconfig \
+            xz
+      - name: Get Dependencies (GTK)
+        if: ${{ needs.what-to-make.outputs.make-gtk == 'true' }}
+        run: apt-get install -y --no-install-recommends libglibmm-2.4-dev libgtkmm-3.0-dev
+      - name: Get Dependencies (Qt)
+        if: ${{ needs.what-to-make.outputs.make-qt == 'true' }}
+        run: apt-get install -y --no-install-recommends qtbase5-dev libqt5svg5-dev qttools5-dev
+      - name: Get Source
+        uses: actions/download-artifact@v3
+        with:
+          name: source-tarball
+      - name: Extract Source
+        run: mkdir src && tar xf transmission*.tar.* -C src --strip-components 1
+      - name: Configure
+        run: |
+          cmake \
+            -S src \
+            -B obj \
+            -G Ninja \
+            -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+            -DCMAKE_INSTALL_PREFIX=pfx \
+            -DCMAKE_PREFIX_PATH=`brew --prefix`/opt/qt@5 \
+            -DENABLE_CLI=${{ (needs.what-to-make.outputs.make-cli == 'true') && 'ON' || 'OFF' }} \
+            -DENABLE_DAEMON=${{ (needs.what-to-make.outputs.make-daemon == 'true') && 'ON' || 'OFF' }} \
+            -DENABLE_GTK=${{ (needs.what-to-make.outputs.make-gtk == 'true') && 'ON' || 'OFF' }} \
+            -DENABLE_MAC=OFF \
+            -DENABLE_QT=${{ (needs.what-to-make.outputs.make-qt == 'true') && 'ON' || 'OFF' }} \
+            -DENABLE_TESTS=OFF \
+            -DENABLE_UTILS=${{ (needs.what-to-make.outputs.make-utils == 'true') && 'ON' || 'OFF' }} \
+            -DENABLE_WEB=${{ (needs.what-to-make.outputs.make-web == 'true') && 'ON' || 'OFF' }} \
+            -DENABLE_WERROR=ON \
+            -DRUN_CLANG_TIDY=OFF
+      - name: Make
+        run: cmake --build obj --config RelWithDebInfo
+      - name: Test
+        if: ${{ needs.what-to-make.outputs.make-tests == 'true' }}
+        env:
+          TMPDIR: /private/tmp
+        run: cmake -E chdir obj ctest -j $(nproc) --build-config RelWithDebInfo --output-on-failure
+      - name: Install
+        run: cmake --build obj --config RelWithDebInfo --target install/strip
+      - uses: actions/upload-artifact@v3
+        with:
+          name: binaries-${{ github.job }}
+          path: pfx/**/*

--- a/.github/workflows/actions.yml
+++ b/.github/workflows/actions.yml
@@ -587,8 +587,8 @@ jobs:
       - name: Get Dependencies
         run: |
           set -ex
-          apt-get update
-          apt-get install -y --no-install-recommends \
+          sudo apt-get update
+          sudo apt-get install -y --no-install-recommends \
             ca-certificates \
             cmake \
             g++ \
@@ -607,10 +607,10 @@ jobs:
             xz-utils
       - name: Get Dependencies (GTK)
         if: ${{ needs.what-to-make.outputs.make-gtk == 'true' }}
-        run: apt-get install -y --no-install-recommends libglibmm-2.4-dev libgtkmm-3.0-dev
+        run: sudo apt-get install -y --no-install-recommends libglibmm-2.4-dev libgtkmm-3.0-dev
       - name: Get Dependencies (Qt)
         if: ${{ needs.what-to-make.outputs.make-qt == 'true' }}
-        run: apt-get install -y --no-install-recommends qtbase5-dev libqt5svg5-dev qttools5-dev
+        run: sudo apt-get install -y --no-install-recommends qtbase5-dev libqt5svg5-dev qttools5-dev
       - name: Get Source
         uses: actions/download-artifact@v3
         with:

--- a/qt/DetailsDialog.cc
+++ b/qt/DetailsDialog.cc
@@ -1389,10 +1389,12 @@ void DetailsDialog::onAddTrackerClicked()
     {
         for (auto const& [ids, urls] : ids_to_urls)
         {
-            torrentSet(
-                torrent_ids_t{ std::begin(ids), std::end(ids) },
-                TR_KEY_trackerAdd,
-                QList<QString>{ std::begin(urls), std::end(urls) });
+            auto urls_list = QList<QString>{};
+            urls_list.reserve(std::size(urls));
+            for (auto const& url : urls)
+                urls_list << url;
+
+            torrentSet(torrent_ids_t{ std::begin(ids), std::end(ids) }, TR_KEY_trackerAdd, urls_list);
         }
     }
 }

